src/pages/Guilds/CreateProposal.tsx

```diff
@@ -16,20 +22,169 @@ const SidebarContent = styled(Box)`
+`;
+
+const Label = styled.span`
+  font-family: Inter;
+  font-style: normal;
+  font-weight: 500;
+  font-size: ${({ size }) => (size ? size : `14px`)};
+  line-height: 20px;
+  display: flex;
+  color: ${({ color }) => (color ? color : `#000000`)};
+  margin-bottom: 0.5rem;
+`;
+
 const CreateProposalPage: React.FC = () => {
+  const history = useHistory();
+  const [editMode, setEditMode] = React.useState(true);
+  const [title, setTitle] = React.useState('');
+  const [referenceLink, setReferenceLink] = React.useState('');
+  const [proposalBodyHTML, setProposalBodyHTML] = React.useState('');
+  const [proposalBodyMd, setProposalBodyMd] = React.useState('');
+
+  const handleToggleEditMode = () => {
+    // TODO: add proper validation if toggle from edit to preview without required fields
+    if (editMode && !title && !proposalBodyMd) return;
+    setEditMode(v => !v);
+  };
+
+  const handleBack = () => history.push('/');
+
+  const handleCreateProposal = () => {
+    // TODO: build this functionality
+    console.log('Create proposal click');
+  };
+
   return (
     <PageContainer>
       <PageContent>
-        <Editor />
+        <Flex
+          direction="row"
+          justifyContent="space-between"
+          margin="0px 0px 24px"
+        >
+          <Button iconLeft onClick={handleBack}>
+            <FiChevronLeft />
+            Back to Overview
+          </Button>
+
+          <Button
+            padding="8px"
+            onClick={handleToggleEditMode}
+            disabled={!title || !proposalBodyMd}
+          >
+            {editMode ? (
+              <MdOutlinePreview size={18} />
+            ) : (
+              <MdOutlineModeEdit size={18} />
+            )}
+          </Button>
+        </Flex>
+        <Box margin="0px 0px 24px">
+          {editMode ? (
+            <>
+              <Label>Title</Label>
+              <Input
+                placeholder="Proposal Title"
+                value={title}
+                onChange={e => setTitle(e.target.value)}
+              />
+            </>
+          ) : (
+            <Label size="24px"> {title}</Label>
+          )}
+        </Box>
+        <Box margin="0px 0px 24px">
+          {editMode ? (
+            <>
+              <Label color="#A1A6B0"> Reference link (optional)</Label>
+
+              <Input
+                placeholder="https://daotalk.org/..."
+                value={referenceLink}
+                onChange={e => setReferenceLink(e.target.value)}
+                icon={<MdLink size={18} color="#BDC0C7" />}
+              />
+            </>
+          ) : referenceLink ? (
+            <Label color="#A1A6B0" size="16px">
+              {referenceLink}
+            </Label>
+          ) : null}
+        </Box>
+        {editMode ? (
+          <Editor
+            onMdChange={setProposalBodyMd}
+            onHTMLChange={setProposalBodyHTML}
+          />
+        ) : (
+          <div dangerouslySetInnerHTML={{ __html: proposalBodyHTML }} />

```

Comment:
Not sure about using `dangerouslySetInnerHTML` here. Doesn't seem to be vulnerable at the first glance, but maybe we can render the Markdown instead? The view proposal page is currently rendering Markdown, so it would also bring more parity to how we render the content.

Reply:
Its fine here so long as we sanitise it